### PR TITLE
[libraries][installer] - ModuleAdapter

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for a partial fix to  Issue #22949 .

### Summary of Changes
cast to int for postgresql

### Testing Instructions
see  #22949

### CODE REVIEW

**Comment by twister65** [comment]
It fixes plugin update issue.
Very strange, the changes are in ModuleAdapter.
The module update issue is still there.

**Comment by twister65** [comment]
Please, replace `$db->quote` by `$db->quoteName`
https://github.com/joomla/joomla-cms/blob/1ea6ca65586baf0608027d87cde9c23a67495ed2/libraries/src/Installer/Adapter/ModuleAdapter.php#L705
`->where($db->quoteName('client_id') . ' = ' . (int) $this->extension->client_id);`

File: libraries/src/Installer/Adapter/ModuleAdapter.php [hunk 1]
```diff
@@ -702,7 +702,7 @@ public function uninstall($id)
 		$query = $db->getQuery(true)
 			->delete($db->quoteName('#__modules'))
 			->where($db->quoteName('module') . ' = ' . $db->quote($this->extension->element))
-			->where($db->quote('client_id') . ' = ' . $this->extension->client_id);
+			->where($db->quote('client_id') . ' = ' . (int) $this->extension->client_id);
```
Comment (Quy) [inline libraries/src/Installer/Adapter/ModuleAdapter.php]:
```suggestion
			->where($db->quoteName('client_id') . ' = ' . (int) $this->extension->client_id);
```

**Review by twister65 — CHANGES_REQUESTED** [review]
I think, this line :
joomla-cms/libraries/src/Installer/Adapter/ModuleAdapter.php

Line 701 in 1ea6ca6

 		$this->extension->delete($this->extension->extension_id); 

should be placed after the query (line 717), when we have finished using it.
It is better to delete the object, when we no longer need it.

**Comment by mbabker** [comment]
@twister65 That should really be a separate pull request instead of lumped in with another unrelated change.

**Comment by twister65** [comment]
See (test) PR #22993 please.

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on 3617699a41effc0448f75335ceb34fe96925cb2f<br /><br />Code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22984">issues.joomla.org/tracker/joomla-cms/22984</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 3617699a41effc0448f75335ceb34fe96925cb2f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22984">issues.joomla.org/tracker/joomla-cms/22984</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22984">issues.joomla.org/tracker/joomla-cms/22984</a>.</sub>

**Pull request merged**